doc/btrfs-multiple-devices.md

```diff
@@ -0,0 +1,101 @@
+
+Multiple Devices Btrfs
+======================
+
+
+Mounting Degraded Btrfs
+-----------------------
+
+Mounting a degraded btrfs requires the 'degraded' mount option.
+
+Using the 'degraded' mount option in /etc/fstab by default was not recommended
+by btrfs experts (on IRC).
+
+To be decided: What should be done during probing with degraded btrfses?
+
+The 'degraded' option is a nop if no devices are missing.
+
+
+Changing RAID Levels
+--------------------
+
+Changing the RAID levels required running btrfs balance job.
+
+  btrfs balance start -dconvert=raid1,soft -mconvert=raid1,soft /test
+
+Changing the RAID levels can require copying huge amounts of data so is
+potentially very time consuming. Usually such operations are not supported by
+libstorage-ng.
+
+As a benefit the balance can be started in the background. But this fails if
+there is already a balance job running (without an error message from the
+btrfs command). The system does not provide a way to queue balance jobs and it
+should not be the task of libstorage-ng to implement such a way.
+
+The balance job can also fail due to missing space. When run in the background
+reporting that to the user is not possible (the balance job starts and simply
+fails later).
+
+If the filesystem needs to be mounted temporarily for the balance job,
+starting the balance job in the background would prevent unmounting. Normally
+this should be a minor problem since the philosophy of btrfs is to be always
+mounted. But during installation this could be problematic.
+
+Thus operations requiring a balance job are not supported.
+
+
+Adding a Device
+---------------
+
+Adding a device to an existing btrfs is easy.
+
+  btrfs device add /dev/sdc1 /test
```

Comment:
how much time consuming it is? E.g. for RAID5 btrfs?

Reply:
I cannot say since I have not proper hardware to test this. All my tests are done in virtual machines. There a balance job for disks with only 8 GiB takes a few minutes. I suppose on real hardware with disks in the TiB range it can take hours,